operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -107,14 +91,12 @@ public void close() {
   }
 
   @Override
-  public void eventReceived(Watcher.Action action, CustomResource customResource) {
+  public void eventReceived(Watcher.Action action, T customResource) {
     log.debug(
         "Event received for action: {}, resource: {}",
         action.name(),
         customResource.getMetadata().getName());
 
-    resourceCache.cacheResource(
-        customResource); // always store the latest event. Outside the sync block is intentional.
```

Comment:
I see you moved this to the DefaultEventHandler. That is not good. Since the custom resource does not gets to the cache if the generations are not changed, this can be a problem in case of a retry for example.

Reply:
The tests are passing, though…

Reply:
Probably we don't have a test for this case.

Reply:
Probably we don't have a test for this case.

Reply:
To be clear, this example shows exactly what I mean: if you can't reason about the behavior of the code without having to take the caching into account then the behavior is fundamentally unsound, in my opinion. If the logic breaks because something wasn't cached, then the logic is wrong.

Reply:
Look @metacosm , I think we discussed this already too much, we agree that this is not ideal, going to be improved/chaged. But it works. With this change it breaks. So we will improve this in a separate issue.